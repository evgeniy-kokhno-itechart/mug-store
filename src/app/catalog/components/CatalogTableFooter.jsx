import React from 'react';
import { PropTypes } from 'prop-types';
import { Dropdown, Pagination } from '../../shared';

const CatalogTableFooter = ({
  totalCount, pageSize, pageSizeOptions, currentPage, onPageChange, onItemsCountChange,
}) => (
  <div className="row justify-content-between mx-0 mx-sm-2">
    <div className="col">
      <Pagination
        itemsCount={totalCount}
        pageSize={pageSize}
        onPageChange={onPageChange}
        currentPage={currentPage}
        pageSizeOptions={pageSizeOptions}
      />
    </div>
    <div className="col">
      <Dropdown
        name="itemsOnPage"
        label="Items on page"
        options={pageSizeOptions}
        value={pageSize.toString()}
        isOnelineElement
        customClasses="justify-content-end"
        onValueChange={onItemsCountChange}
      />
    </div>
  </div>
);

CatalogTableFooter.propTypes = {
  totalCount: PropTypes.number.isRequired,
  pageSize: PropTypes.number.isRequired,
  pageSizeOptions: PropTypes.arrayOf(PropTypes.shape({ id: PropTypes.string, name: PropTypes.string })).isRequired,
  currentPage: PropTypes.number.isRequired,
  onPageChange: PropTypes.func.isRequired,
  onItemsCountChange: PropTypes.func.isRequired,
};

export default CatalogTableFooter;
