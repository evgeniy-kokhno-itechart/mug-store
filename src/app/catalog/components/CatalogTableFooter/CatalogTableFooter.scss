@import "Variables";

.catalog__footer {
    display: flex;
    flex-wrap: wrap;
    align-items: center;
    margin: 1rem 1rem 0;
}

.page-links{
    flex: 1;
}

.page-size {
    flex: 1;
    justify-content: flex-end;

    &__dropdown {
        padding: $padding-vertical;
    }
}