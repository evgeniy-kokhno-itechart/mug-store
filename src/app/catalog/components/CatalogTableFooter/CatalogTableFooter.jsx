import React from 'react';
import { PropTypes } from 'prop-types';
import { DropdownInline, Pagination } from '../../../shared';
import './CatalogTableFooter.scss';

const CatalogTableFooter = ({
  totalCount, pageSize, pageSizeOptions, currentPage, onPageChange, onItemsCountChange,
}) => (
  <div className="catalog__footer">
    <Pagination
      itemsCount={totalCount}
      pageSize={pageSize}
      wrapperClasses="page-links"
      onPageChange={onPageChange}
      currentPage={currentPage}
      pageSizeOptions={pageSizeOptions}
    />
    <DropdownInline
      name="itemsOnPage"
      label="Items on page"
      options={pageSizeOptions}
      value={pageSize.toString()}
      wrapperClasses="page-size"
      selectClasses="page-size__dropdown"
      onChange={onItemsCountChange}
    />
  </div>
);

CatalogTableFooter.propTypes = {
  totalCount: PropTypes.number.isRequired,
  pageSize: PropTypes.number.isRequired,
  pageSizeOptions: PropTypes.arrayOf(PropTypes.shape({ id: PropTypes.string, name: PropTypes.string })).isRequired,
  currentPage: PropTypes.number.isRequired,
  onPageChange: PropTypes.func.isRequired,
  onItemsCountChange: PropTypes.func.isRequired,
};

export default CatalogTableFooter;
