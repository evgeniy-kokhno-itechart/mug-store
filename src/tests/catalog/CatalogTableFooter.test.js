/* eslint-disable no-undef */
import React from 'react';

import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import CatalogTableFooter from '../../catalog/components/CatalogTableFooter';

configure({ adapter: new Adapter() });

describe('<CatalogTableFooter />', () => {
  let CatalogTableFooterWrapper;

  beforeEach(() => {
    const onPageChange = jest.fn();
    const onItemsCountChange = jest.fn();
    CatalogTableFooterWrapper = shallow(
      <CatalogTableFooter
        totalCount={3}
        pageSize={10}
        pageSizeOptions={[{ id: '5', name: '5' }]}
        currentPage={1}
        onPageChange={onPageChange}
        onItemsCountChange={onItemsCountChange}
      />,
    );
  });

  it('check render properly', () => {
    expect(CatalogTableFooterWrapper.debug()).toMatchSnapshot();
  });
});
