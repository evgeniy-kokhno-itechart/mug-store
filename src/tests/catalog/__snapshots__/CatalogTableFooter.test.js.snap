// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CatalogTableFooter /> check render properly 1`] = `
<div
  className="row justify-content-between mx-0 mx-sm-2"
>
  <div
    className="col"
  >
    <Pagination
      currentPage={1}
      itemsCount={3}
      onPageChange={[MockFunction]}
      pageSize={10}
      pageSizeOptions={
        Array [
          Object {
            "id": "5",
            "name": "5",
          },
        ]
      }
    />
  </div>
  <div
    className="col"
  >
    <Dropdown
      customClasses="justify-content-end"
      defaultText=""
      error=""
      isOnelineElement={true}
      label="Items on page"
      name="itemsOnPage"
      onValueChange={[MockFunction]}
      options={
        Array [
          Object {
            "id": "5",
            "name": "5",
          },
        ]
      }
      value="10"
    />
  </div>
</div>
`;
